crates/kitsune_p2p/types/src/tx2/tx2_utils/share.rs

```diff
@@ -0,0 +1,66 @@
+use crate::*;
+
+/// Synchronized droppable share-lock around internal state date.
+pub struct Share<T: 'static + Send>(Arc<parking_lot::Mutex<Option<T>>>);
```

Comment:
This feels like the new ghost actor :)

Reply:
totally - but so short it doesn't deserve a whole crate of its own : )